hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroIndexedRecord.java

```diff
@@ -153,29 +152,11 @@ public HoodieRecord wrapIntoHoodieRecordPayloadWithParams(
       Option<Pair<String, String>> simpleKeyGenFieldsOpt,
       Boolean withOperation,
       Option<String> partitionNameOp,
-      Boolean populateMetaFields) {
+      Boolean populateMetaFields,
+      Option<Schema> schemaWithoutMetaFields) {
     String payloadClass = ConfigUtils.getPayloadClass(props);
     String preCombineField = ConfigUtils.getOrderingField(props);
-    return HoodieAvroUtils.createHoodieRecordFromAvro(data, payloadClass, preCombineField, simpleKeyGenFieldsOpt, withOperation, partitionNameOp, populateMetaFields);
-  }
-
-  public HoodieRecord wrapIntoHoodieRecordPayloadWithoutMetaFields(
-      Schema recordSchema,
-      Schema schemaWithoutMetaFields,
-      Properties props,
-      Option<Pair<String, String>> simpleKeyGenFieldsOpt,
-      Boolean withOperation,
-      Option<String> partitionNameOp,
-      Boolean populateMetaFields) {
-    String payloadClass = ConfigUtils.getPayloadClass(props);
-    String preCombineField = ConfigUtils.getOrderingField(props);
-    return SpillableMapUtils.convertToHoodieRecordPayload2((GenericRecord) data,
-        payloadClass,
-        preCombineField,
-        simpleKeyGenFieldsOpt.orElse(Pair.of(HoodieRecord.RECORD_KEY_METADATA_FIELD, HoodieRecord.PARTITION_PATH_METADATA_FIELD)),
-        withOperation,
-        partitionNameOp,
-        schemaWithoutMetaFields);
+    return HoodieAvroUtils.createHoodieRecordFromAvro(data, payloadClass, preCombineField, simpleKeyGenFieldsOpt, withOperation, partitionNameOp, populateMetaFields, schemaWithoutMetaFields);
```

Comment:
So, do we have any follow ups for SparkRecord to work ? I see we made changes to avro based here.

Reply:
spark record was updated accordingly https://github.com/apache/hudi/pull/8490/commits/2fa5dfe7b47172dd9cf342c79d4938bc92a6f7e0#diff-247de102c8e4aa9857e70ed8f77e306a87005c6d863f86de3b69610f2a696457